macros/get_lookback_date_limits.sql

```diff
@@ -33,22 +57,65 @@
     {% endif %}
   {% endif %}
 
+  -- setting and executing the limit query depending on input
 
   {% set query %}
-    {% if limit_type == 'min' %}
-      with base as (select case when '{{ var("snowplow__conversion_window_start_date") }}' = ''
-                  then {{ dbt.dateadd('day', -31, dbt.current_timestamp()) }}
-                  else '{{ var("snowplow__conversion_window_start_date") }}'
-                  end as min_date_time)
-      select cast({{ dbt.dateadd('day', (- var('snowplow__path_lookback_days') + 1), 'min_date_time') }} as date) from base
-
-
-    {% elif limit_type == 'max' %}
-      with base as (select case when '{{ var("snowplow__conversion_window_start_date") }}' = ''
-                  then {{ dbt.dateadd('day', -1, dbt.current_timestamp()) }}
-                  else '{{ var("snowplow__conversion_window_end_date") }}'
-                  end as max_date_time)
-      select cast(max_date_time as date) from base
+    {% if limit_type == 'min' and model == 'sessions' %}
+      {% if var("snowplow__conversion_window_start_date") == '' %}
+          with base as (
+            select max(start_tstamp) as last_pageview
+            from {{ var('snowplow__page_views_source') }}
+          )
+          select cast({{ dbt.dateadd('day', -combined_time, 'last_pageview') }} as date) as lower_limit
+          from base
+
+      {% else %}
+         with base as (
+          select cast('{{ var('snowplow__conversion_window_start_date')}}' as timestamp) as cw_tstamp
+         )
+        select cast({{ dbt.dateadd('day', -var('snowplow__path_lookback_days'), 'cw_tstamp' ) }} as date) as lower_limit
+        from base
+      {% endif %}
+
+    {% elif limit_type == 'max' and model == 'sessions' %}
+      {% if var("snowplow__conversion_window_start_date") == '' %}
+          with base as (
+                select max(start_tstamp) as last_pageview
```

Comment:
end date?

Reply:
I was not fully explicit here. In the above step we rule out cases where either start date or end date is given but not the other, it will error out. Technically it is fine at this stage to just check if only the start date is missing but I will add both conditions again just so it doesn't confuse developers.